src/pudl/dagster_etl.py

```diff
@@ -0,0 +1,81 @@
+"""Dagster version of EPA CEMS ETL."""
+from pathlib import Path
+
+import pandas as pd
+import sqlalchemy as sa
+from dagster import (AssetMaterialization, EventMetadata, Field, In, Output,
+                     job, op, resource)
+
+import pudl
+from pudl.helpers import compute_dataframe_summary_statistics
+from pudl.metadata.classes import Resource
+from pudl.workspace.datastore import Datastore
+
+# TODO: how can we associate metadata schemas with specific ops?
+EpaCemsDataFrame = Resource.from_id("hourly_emissions_epacems").to_dagster(
+    event_metadata_fn=compute_dataframe_summary_statistics)
+
+
+@op(required_resource_keys={"pudl_settings"}, ins={"transformed_df": In(dagster_type=EpaCemsDataFrame)})
+def load_epacems(context, transformed_df: pd.DataFrame):
+    """Load epacems to parquet."""
+    root_path = Path(context.resources.pudl_settings["parquet_dir"]) / "epacems"
+
+    pudl.load.df_to_parquet(
+        transformed_df,
+        resource_id="hourly_emissions_epacems",
+        root_path=root_path,
+        partition_cols=["year", "state"]
+    )
+
+    # Format of the mapping key makes it difficult to recreate the final parquet path.
+    yield AssetMaterialization(
+        asset_key="hourly_emissions_epacems",
+        description="Persisted result to storage",
+        partition=context.get_mapping_key(),
+        metadata={
+            "Description": f"Parquet file of {context.get_mapping_key()} EPA CEMS data.",
+            "Path": EventMetadata.path(str(root_path)),
+            "Number of Rows": len(transformed_df)
+        },
+    )
+    yield Output(root_path)
+
+# The configurable @resources the ETL depends on.
+
+
+@resource
+def pudl_settings(init_context):
```

Comment:
`init_context` is a Dagster keyword that contains Resource information.

I made `pudl_settings` a resource so it can be shared across multiple ops in a the job.

Reply:
It looks like there's this function [`make_values_resources()`](https://docs.dagster.io/concepts/configuration/config-schema#passing-configuration-to-multiple-ops-in-a-job) specifically for packing a config into a resource so you can share it across all the ops in a job. Not sure I totally understand what it's doing though.

Reply:
Oh nice find. `make_values_resources()` seems useful since we specify config for a DAG instead of specific ops.